WordPress/gutenberg
# Restore default border and focus style on image url input field.

Fixes https://github.com/WordPress/gutenberg/issues/58494

## What?

The UI to inseert link on images is different from the default URL input UI and has now an insufficient, and inconsistent, focus style. This UI is used in a few blocks e.g. Image, Media & Text, etc.

## Why?

Focus style must always be clear. Ideally it should be consistent across all the input fields in the editor UI.
There's no good reason why the image URL input mut look and work differently from the default URL input UI used for text, besides very specific features that are only available for images.

This PR aims to make the image URL input use the default focus style. To do so, some additional CSS changes are necessary like adjusting the input and button sizes to the new defaults and better spacing and alignment. I tried to keep the changest o a minimum. 

There's more inconsistencies to fix, both visual and functional. I'll create a follow=up issue.

## How?

- Adjusts the CSS to use the default focus style for input fields.
- Restores the input border: the border is integrant part of the focus style and should never be overridden with 'ad hoc' CSS.
- Adjust input and button sizes.
- Adjusts spacing around the input and alignments.

## Testing Instructions





### Testing Instructions for Keyboard

- HInt: compare with the 'default' URL input UI used for links on text content.
- Add a link to an image for example in a Media & Text block or an Image block.
- The image URL input popover opens.
- Initial focus is set on the input field.
- Observe the focus style is the default one used for all input fields.
- Observe the input field and buttons sizes is nor bigger.
- Observe spacing and alignments are closer to the ones of the default URL input UI.

## Screenshots or screencast 

Before: non focused and focused:

![before 01](https://github.com/WordPress/gutenberg/assets/1682452/48fc211a-1fd1-4a50-afea-b9bd7ca8492d)

After: non focused and focused:

![after 01](https://github.com/WordPress/gutenberg/assets/1682452/6b060b27-1a04-43fa-aabe-7fc999b1e2a3)

Compare with the default URL input UI:

![default 01](https://github.com/WordPress/gutenberg/assets/1682452/c0f723b3-380c-4863-a86b-02ec09e5706f)
